src/structures/ActionRow.js

```diff
@@ -0,0 +1,104 @@
+'use strict';
+
+const BaseComponent = require('./BaseComponent');
+const Components = require('../ComponentFactory');
+const { ComponentTypes } = require('../util/Constants');
+
+/**
+ * Represents an action row containing components.
+ * @extends {BaseComponent}
+ */
+class ActionRow extends BaseComponent {
+  /**
+   * Components that can be placed in an action row
+   * * Button
+   * * SelectMenu
+   * * InputText
+   * @typedef {Button|SelectMenu|InputText} ActionRowComponent
+   */
+
+  /**
+   * Options for components that can be placed in an action row
+   * * ButtonOptions
+   * * SelectMenuOptions
+   * * InputTextOptions
+   * @typedef {ButtonOptions|SelectMenuOptions|InputTextOptions} ActionRowComponentOptions
+   */
+
+  /**
+   * Data that can be resolved into components that can be placed in an action row
+   * * ActionRowComponent
+   * * ActionRowComponentOptions
+   * @typedef {ActionRowComponent|ActionRowComponentOptions} ActionRowComponentResolvable
+   */
+
+  /**
+   * @typedef {BaseComponentOptions} ActionRowOptions
+   * @property {ActionRowComponentResolvable[]} [components]
+   * The components to place in this action row
+   */
+
+  /**
+   * @param {ActionRow|ActionRowOptions} [data={}] ActionRow to clone or raw data
+   * @param {Client} [client] The client constructing this ActionRow, if provided
+   */
+  constructor(data = {}, client = null) {
+    super({ type: 'ACTION_ROW' });
+
+    /**
+     * The components in this action row
+     * @type {ActionRowComponent[]}
+     */
+    this.components = data.components?.map(c => Components.create(c, client)) ?? [];
+  }
+
+  /**
+   * Adds components to the action row.
+   * @param {...ActionRowComponentResolvable[]} components The components to add
+   * @returns {ActionRow}
+   */
+  addComponents(...components) {
+    this.components.push(...components.flat(Infinity).map(c => Components.create(c)));
```

Comment:
```suggestion
    this.components.push(...components.flat(Infinity).map(c => Components.create(c, client)));
```

Reply:
Where does client come from, Rodry?

Reply:
The constructor parameter, kyra.

Reply:
Actually, no, client shouldn't even be passed down here. It will make the error be emitted as a debug message instead of thrown